### PR TITLE
Split data creation and update into two methods

### PR DESCRIPTION
**Pull request by perlpunk**
This makes the methods shorter and allows simpler unit tests in the future

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm [hunk 1]
```diff
@@ -491,6 +432,72 @@ sub update {
     $self->respond_to(json => {json => $json});
 }
 
+sub create_job_templates_from_yaml {
```
Comment (kraih) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm]: Why is this a public method?

File: lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm [hunk 2]
```diff
@@ -491,6 +432,72 @@ sub update {
...
+    my ($self, $yaml) = @_;
```
Comment (kraih) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm]: `$self` does not appear to be used at all.

**Comment by kraih** [comment]
Controller methods (or actions if public, as is the case here) are not really that good for unit testing.

**Comment by perlpunk** [comment]
> Controller methods (or actions if public, as is the case here) are not really that good for unit testing.

But do you agree that splitting this is a good idea or do we not aim for short methods and less nesting?

If you think splitting is a good idea, where should I put it?

**Comment by kraih** [comment]
Splitting it seems reasonable. Looking at the surrounding code, most of this probably belongs in the model anyway.

**Comment by perlpunk** [comment]
> Splitting it seems reasonable. Looking at the surrounding code, most of this probably belongs in the model anyway.

I can rename the method with an underscore for now.
I thought it would be a good idea and a small step to untangle the code, and we can then move it to a better place later.
But @kalikiana is also working on this code, so I don't know how we should proceed here.

**Comment by kraih** [comment]
I'm fine with an underscore function for now, but the code would be better off in a model class.

**Comment by perlpunk** [comment]
I added an underscore and removed the `$self`

**Pull request merged**